File: Y2S2/SDI/lab5-8/packageManagerLab8/core/src/main/java/ro/arc/packageManager/core/service/MaintainerService.java
package ro.arc.packageManager.core.service;

import ro.arc.packageManager.core.domain.Maintainer;

import java.util.List;

public interface MaintainerService {
    Maintainer addMaintainer(Maintainer maintainer);

    Maintainer updateMaintainer(Maintainer maintainer);

    void deleteMaintainer(Long ID);

    List<Maintainer> getFilteredMaintainers(String type, String input);

    List<Maintainer> getAllMaintainers();
}

File: Y2S2/SDI/lab9-11/packageManagerLab10/web/src/main/webapp/src/app/maintainers/shared/maintainer.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs";
import { MaintainerModel } from './maintainer.model';

@Injectable()
export class MaintainerService {
  private maintainersUrl = 'http://localhost:8080/web-1.0-SNAPSHOT/api/maintainers';

  constructor(private httpClient: HttpClient) { }

  getMaintainers(): Observable<MaintainerModel[]> {
    return this.httpClient.get<Array<MaintainerModel>>(this.maintainersUrl);
  }

  addMaintainer(maintainer: MaintainerModel): Observable<MaintainerModel> {
    return this.httpClient.post<MaintainerModel>(this.maintainersUrl, maintainer);
  }

  updateMaintainer(maintainer: MaintainerModel): Observable<MaintainerModel> {
    return this.httpClient.put<MaintainerModel>(`${this.maintainersUrl}/${maintainer.id}`, maintainer);
  }

  deleteMaintainer(maintainer: MaintainerModel): Observable<any> {
    return this.httpClient.delete(`${this.maintainersUrl}/${maintainer.id}`);
  }
}
